reflex/components/component.py

```diff
@@ -286,7 +299,7 @@ def __init__(self, *args, **kwargs):
                     if kwargs[key] is None:
                         raise TypeError
 
-                    expected_type = fields[key].outer_type_.__args__[0]
+                    expected_type = field_type.__args__[0]
```

Comment:
we should probably change this to `typing.get_args(field_type)[0]` while refactoring